include/seqan3/range/container/small_string.hpp

```diff
@@ -0,0 +1,265 @@
+// -----------------------------------------------------------------------------------------------------
+// Copyright (c) 2006-2019, Knut Reinert & Freie Universität Berlin
+// Copyright (c) 2016-2019, Knut Reinert & MPI für molekulare Genetik
+// This file may be used, modified and/or redistributed under the terms of the 3-clause BSD-License
+// shipped with this file and also available at: https://github.com/seqan/seqan3/blob/master/LICENSE.md
+// -----------------------------------------------------------------------------------------------------
+
+/*!\file
+ * \author Svenja Mehringer <svenja.mehringer AT fu-berlin.de>
+ * \brief A constexpr string implementation to manipulate string literals at compile time.
+ */
+
+#pragma once
+
+#include <seqan3/range/container/small_vector.hpp>
+
+namespace seqan3
+{
+
+/*!\brief Implements a small string that can be used for compile time computations.
+ * \ingroup container
+ * \tparam N The size of the small string.
+ *
+ * This class provides a string type that can be constructed, evaluated and operated on at compile time.
+ * The stored string can be accessed as either a std::string or a c-style string through the respective member functions.
+ * All operations are constexpr except the conversion functions.
```

Comment:
What do you think about:

This class provides a string type for small strings and compile-time contexts. It has fixed capacity, but variable size within the capacity. It is always allocated on the stack and most of it's members are `constexpr`-qualified. The underlying data can be exposed as a null-terminated c-style string (without copying) and conversion operators to std::string are provided (this involves copying).